### ISSUE TITLE
openssl 1.1.0 support for latest debian releases (c++)

### ISSUE THREAD

**Issue by asaiddd** [ftexchange/ftx#77]
Hi, when I am trying to compile rest c;lient, I get an error related to libssl package as below in Ubuntu 20.04.

```
ftx/util/Encoding.h:21:17: error: ‘HMAC_CTX_init’ was not declared in this scope
     HmacCtx() { HMAC_CTX_init(&ctx); }
                 ^~~~~~~~~~~~~
ftx/util/Encoding.h:21:17: note: suggested alternative: ‘HMAC_CTX_new’
     HmacCtx() { HMAC_CTX_init(&ctx); }
                 ^~~~~~~~~~~~~
                 HMAC_CTX_new
ftx/util/Encoding.h: In destructor ‘ftx::util::encoding::HmacCtx::~HmacCtx()’:
ftx/util/Encoding.h:22:18: error: ‘HMAC_CTX_cleanup’ was not declared in this scope
     ~HmacCtx() { HMAC_CTX_cleanup(&ctx); }
                  ^~~~~~~~~~~~~~~~
ftx/util/Encoding.h:22:18: note: suggested alternative: ‘HMAC_CTX_get_md’
     ~HmacCtx() { HMAC_CTX_cleanup(&ctx); }
                  ^~~~~~~~~~~~~~~~
```

This  [answer](https://stackoverflow.com/a/63153643) explains that libssl now does not support allocating HMAC_CTX struct in stack. You should use pointer instead and handle accordingly. As a newcomer I don't feel powerful enough to fix this. Can you please update to support newer libssl versions ?

Thanks.
A.Sait

**Comment by kozyilmaz** [comment]
https://github.com/ftexchange/ftx/pull/13

**Comment by asaiddd** [comment]
Hi @kozyilmaz  just seen your pull request and I was trying it. Thank you for your contribution!

**Comment by asaiddd** [comment]
#13 resolves the compatibilty issue with openssl 1.1

**Comment by rajhlinux** [comment]
How can I get the fix for this?
